libr/core/cmd_zign.c

```diff
@@ -731,9 +731,80 @@ struct ctxSearchCB {
 	const char *prefix;
 };
 
+static char *types_list_to_fcnstr(RList *types) {
+	char *type_kv = NULL, *k = NULL, *v = NULL;
+	char *field = NULL, *name = NULL, *rettype = NULL;
+	char **args, *ret = NULL;
+	int nargs = 0, i = 0, j = 0;
+	RListIter *iter;
+
+	r_list_foreach(types, iter, type_kv) {
+		k = strtok (type_kv, "=");
+		v = strtok (NULL, "\0");
+
+		strtok (k, ".");
+		name = strtok (NULL, ".");
+		field = strtok (NULL, ".");
+
+		while (strcmp (field, "ret") && strcmp (field, "args") &&
+		  strcmp (field, "arg") && strcmp (field, r_str_newf ("%d", i))) {
```

Comment:
* dont indent with spaces.
* this condition is quite complex and will be better to move into a separate function like `while (__validField())`
* the r_str_newf is leaking